src/arrow_interface/arrow_schema.cpp

```diff
@@ -96,4 +104,105 @@ namespace sparrow
         target.release = release_arrow_schema;
     }
 
+    bool check_compatible_schema(const ArrowSchema& schema1, const ArrowSchema& schema2)
+    {
+        if (&schema1 == &schema2)
+        {
+            return true;
+        }
+        if (schema1.release == nullptr || schema2.release == nullptr)
+        {
+            return false;
+        }
```

Comment:
This is slightly inconsistent: a schema is equal to itself through pointer identity, but not to a bit-identical schema.